### PR TITLE
Remote operation of loader

### PR DESCRIPTION
**Pull request by wpears**
Allows the loader to run fairly robustly on S3 buckets and remote zips/geojson.
Data is streamed as much as is feasible, with some workarounds for ogr2ogr.

Last ~6 commits are silly travis integration marginalia

### CODE REVIEW

**Comment by wpears** [comment]
Paging @hkeeler, how's this looking?

**Comment by hkeeler** [comment]
@wpears, reviewing now.  Will have all comments up and probably merged before tomorrow's standup.

**Comment by hkeeler** [comment]
There's a fair amount of duplication across:
* [`.travis.yml:8-23`](https://github.com/cfpb/grasshopper-loader/pull/52/files#diff-354f30a63fb0907d4ad57269548329e3R8)
* [`buildsteps`](https://github.com/cfpb/grasshopper-loader/pull/52/files#diff-d98acfbe524ef2e2b91cdf1db642ddf1R1)
* [`install_gdal_ubuntu.sh`](https://github.com/cfpb/grasshopper-loader/pull/52/files#diff-d4541c0293adf5989ace9109395df9c9R1)

Seems like much of that could be merged.  Also, a few questions:
* Is `buildsteps` intended to be executed as a script, or is it more instructional?
* Do we want to use Travis's built-in Elasticsearch and then our own build in our env?  Worried about versions drifting apart.

Not going to hold up the merge on this, but I think we should discuss further.  Also, not sure how all this setup plays into Docker.  Just want to reduce as much duplication as different versions across environments could be a headache down the line.

**Comment by wpears** [comment]
buildsteps was intended as a more instructional set of commands from which different shell scripts would be written (of which install_gdal_ubuntu.sh) is an example. In retrospect, this is probably not the best name for the file, since it also installs node.. I'll add a commit changing this.

As well, I believe I was having some peculiar issue with using the install_gdal_ubuntu.sh as a preinstall script in Travis, which is why there is duplication. I'll attempt to rectify this as well.

And to the Docker question, I'll be answering it in another PR soon.

File: lib/unzipGeoStream.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+var path = require('path');
+var mkdirp = require('mkdirp');
+var unzip = require('unzip');
+var rmrf = require('rimraf');
+
+
+function unzipGeoStream(zip, stream, counter, geoCb, finalCb){
+  var dirname = path.join(path.dirname(zip), Math.round(Math.random()*1e15).toString(), path.basename(zip, '.zip'));
```
Comment (hkeeler) [inline lib/unzipGeoStream.js]: What's going on here?  Are you generating a random directory name?  I assume so multiple runs of the loader won't unzip the file to the same destination?  A comment for the intent would be helpful, perhaps with an example of what the directory name should look like.

File: lib/transformerTemplate.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ module.exports = function(addr, city, state, zip){
                        ),
           alt_address: "",
           load_date: Date.now()
+          //Source<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```
Comment (hkeeler) [inline lib/transformerTemplate.js]: Did you mean to leave this here?

**Comment by hkeeler** [comment]
@wpears, the changes to the build stuffs looks good.  Interesting you hit up against the `sudo: true` issue in Travis.  I was just reading yesterday about how `sudo: false` puts you on their [Docker-based infrastructure](http://docs.travis-ci.com/user/workers/container-based-infrastructure/), which makes things very speedy, but you lose much of your super powers.

Overall, everything looks OK.  I put up a couple questions related to commenting stuff, but that doesn't need to hold things up.  Just want to confirm you don't have any other changes for this PR before I hit "Merge".

**Comment by wpears** [comment]
Yeah looks fine to merge. Travis tries to pick up whether you need sudo even if you don't set sudo: true, but apparently they don't check scripts you're running, just the .travis.yml file. So moving some sudo expressions out of it caused the trouble.

**Pull request merged**